math/pythagoras.cpp

```diff
@@ -0,0 +1,119 @@
+/**
+ * Copyright 2020 @author beqakd
+ * @file
+ *
+ * A basic implementation of 3d distance between two point.
+ */
+
+#include <cassert>   // For testing(assert)
+#include <cmath>     // For std::pow(), std::sqrt() and so on
+#include <iostream>  // For io operations
+
+/**
+ * The distance between two points in a three dimensional - 3D - coordinate
+ * system. For this we have struct point where we save coordinates of our points
+ * and later on calculate the distance as mentioned.
+ */
+class Distance {
+ public:
+    /**
+    * Point struct to save 3D coordinates.
+    */
+    struct Point {
+        int x;
+        int y;
+        int z;
+    };
+
+    // Our two points to calculate distance.
+    Point a;
```

Comment:
no need to declare the member variable inside the class just implement the `class point` and create the instances of point class in `test()` to declare two points and call the `point distance(const point &, const point &)` member function on those points this function will return the distance between two points